haskala/modules/custom/haskala_book/haskala_book.module

```diff
@@ -17,24 +17,81 @@ function haskala_book_ctools_plugin_directory($module, $plugin) {
 
 /**
  *  Prefaces tab - role in book production of preface writer.
+ *  Book producers tab - all fields from production bundle.
  *
  *  @param $book_nid
- *   The book nid.
+ *   The book node ID.
  *  @param $person_nid
- *   The writer person nid.
+ *   The writer person node ID.
  *
- *  @return production details nid.
+ *  @return production details nodes.
  */
-function haskala_book_get_production($book_nid, $person_nid) {
+function haskala_book_get_production($book_nid, $person_nid = NULL) {
   $query = new EntityFieldQuery();
   $query->entityCondition('entity_type', 'node')
     ->entityCondition('bundle', 'production')
     ->propertyCondition('status', NODE_PUBLISHED)
-    ->fieldCondition('field_producer', 'target_id', $person_nid)
     ->fieldCondition('field_book', 'target_id', $book_nid);
+  if ($person_nid) {
+    $query->fieldCondition('field_producer', 'target_id', $person_nid);
+  }
   $result = $query->execute();
   if (empty($result['node'])) {
     return;
   }
-  return reset($result['node'])->nid;
+  // If filter by person also, we should get single result.
+  if ($person_nid) {
+    return node_load(reset($result['node'])->nid);
+  }
+  return node_load_multiple(array_keys($result['node']));
 }
+
+
+/**
+ *  Get tab group fields.
+ *
+ *  @param $wrapper
+ *   The wrapper object.
+ *  @param $tab
+ *   The tab info.
+ *  @param $node_view
+ *   The node view array for weight field info.
+ * @param $book_nid
+ *   The book nid.
+ *
+ *  @return tab fields array.
+ */
+function haskala_book_get_tab_fields($wrapper, $tab, $node_view, $book_nid) {
+
+  // Sort fields by weight.
+  $tab_fields_sorted = array();
+  foreach($tab->children as $field_name) {
+    if ($wrapper->$field_name->value()){
+      $weight = $node_view[$field_name]['#weight'];
+      $tab_fields_sorted[$weight] = $field_name;
+    }
+  }
+  ksort($tab_fields_sorted);
+
+  $tab_fields = array();
+  foreach($tab_fields_sorted as $field_name) {
+
+    // Check if the field isn't empty.
+    if ($wrapper->$field_name->value()) {
+      $field = render($node_view[$field_name]);
+      $tab_fields[] = $field;
+
+      // Role of preface writer.
+      if ($field_name == 'field_writer_of_preface') {
+        $production_node = haskala_book_get_production($book_nid, $wrapper->$field_name->value()->nid);
```

Comment:
Yes. Several roles are in single field of multi roles terms (Occupation).
![image](https://cloud.githubusercontent.com/assets/123744/5915030/757f508a-a608-11e4-8a52-a0966d196626.png)